tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -2605,6 +2606,248 @@ public static void test_supportsLikeEscapeClause_shouldBe_true() throws Exceptio
+		final String CATALOG_NAME = "tmp";
+		final String TABLE_NAME = "t1";
+		final String qualifiedTableName = CATALOG_NAME + "." + TABLE_NAME;
+		ResultSet resultSet = null;
+		try {
+			final File file = Files.createTempFile("duckdb-jdbc-test-", ".duckdb").toFile();
+			file.delete();
+			statement.execute("ATTACH '" + file.getAbsolutePath() + "' AS \"" + CATALOG_NAME + "\"");
+			file.deleteOnExit();
+			
+			final boolean supportsCatalogsInTableDefinitions = databaseMetaData.supportsCatalogsInTableDefinitions();
+			try {
+				statement.execute("CREATE TABLE " + qualifiedTableName + "(id int)");
+			}
+			catch (SQLException ex) {
+				if (supportsCatalogsInTableDefinitions) {
+					fail(
+						"supportsCatalogsInTableDefinitions is true but CREATE TABLE in attached database is not allowed. "+
+						ex.getMessage()
+					);
+					ex.printStackTrace();
+				}
+			}
+			resultSet = statement.executeQuery(
+				"SELECT * FROM information_schema.tables "+
+				"WHERE table_catalog = '" + CATALOG_NAME + "' " +
+				"AND table_name = '" + TABLE_NAME + "' "
+			);
+			assertTrue(resultSet.next(), "Expected exactly 1 row from information_schema.tables, got 0");
+			assertFalse(resultSet.next());
+			resultSet.close();
+
+			try {
+				statement.execute("DROP TABLE " + qualifiedTableName);
+			}
+			catch (SQLException ex) {
+				if (supportsCatalogsInTableDefinitions) {
+					fail(
+						"supportsCatalogsInTableDefinitions is true but DROP TABLE in attached database is not allowed. " + 
+						ex.getMessage()
+					);
+					ex.printStackTrace();
+				}
+			}
+			resultSet = statement.executeQuery(
+				"SELECT * FROM information_schema.tables "+
+				"WHERE table_catalog = '" + CATALOG_NAME + "' " +
+				"AND table_name = '" + TABLE_NAME + "' "
+			);
+			assertTrue(resultSet.next() == false, "Expected exactly 0 rows from information_schema.tables, got > 0");
+			resultSet.close();
+
+			assertTrue(supportsCatalogsInTableDefinitions, "supportsCatalogsInTableDefinitions should return true.");
+		} 
+		finally {
+			if (resultSet != null) {
+				resultSet.close();
+			}
+			statement.close();
+			connection.close();
+		}
+	}
+
+	public static void test_supports_catalogs_in_data_manipulation() throws Exception {
+		final File file = Files.createTempFile("duckdb-jdbc-test-", ".duckdb").toFile();
+		final String filePath = file.getAbsolutePath();
+		final String CATALOG_NAME = "tmp";
+		final String TABLE_NAME = "t1";
+		final String COLUMN_NAME = "id";
+		final String qualifiedTableName = CATALOG_NAME + "." + TABLE_NAME;
+		file.delete();
+		Connection connection = null;
+		Statement statement = null;
+		
+		connection = DriverManager.getConnection("jdbc:duckdb:" + filePath);
+		file.deleteOnExit();
+		statement = connection.createStatement();
+		statement.execute("CREATE TABLE " + TABLE_NAME + "(" + COLUMN_NAME + " int)");
+		statement.close();
+		connection.close();
+		
+		Thread.currentThread().sleep(1000);
```

Comment:
Is it necessary for this test to attach an on-disk database (and hence create challenges around locks and such)? Perhaps we could just attach an in-memory database in the test?

Reply:
> Thanks for the PR! The changes to the driver seem fine to me - but the tests seem a bit too involved/complex. Could we perhaps simplify the tests and use an in-memory database in `ATTACH`? That should test the same properties but avoid the need for sleeps and sidestep file creation/locking issues.

Yeah good call. Tests modified. 
https://github.com/duckdb/duckdb/pull/6239/commits/a0327f2a8f089073702a03f0d111edf793cb69f1